src/main/java/org/cryptomator/macos/keychain/TouchIdKeychainAccess.java

```diff
@@ -0,0 +1,75 @@
+package org.cryptomator.macos.keychain;
+
+import org.cryptomator.integrations.common.OperatingSystem;
+import org.cryptomator.integrations.common.Priority;
+import org.cryptomator.integrations.keychain.KeychainAccessException;
+import org.cryptomator.integrations.keychain.KeychainAccessProvider;
+import org.cryptomator.macos.common.Localization;
+
+/**
+ * Stores passwords in the macOS system keychain. Requires an authenticated user to do so.
+ * Authentication is done via TouchID or password as a fallback, when TouchID is not available.
+ * <p>
+ * Items are stored in the default keychain with the service name <code>Cryptomator</code>, unless configured otherwise
+ * using the system property <code>cryptomator.integrationsMac.keychainServiceName</code>.
+ */
+@Priority(1000)
+@OperatingSystem(OperatingSystem.Value.MAC)
+public class TouchIdKeychainAccess implements KeychainAccessProvider {
+
+	private static final String SERVICE_NAME = System.getProperty("cryptomator.integrationsMac.keychainServiceName", "Cryptomator");
+
+	private final MacKeychain keychain;
+
+	public TouchIdKeychainAccess() {
+		this(new MacKeychain());
+	}
+
+	// visible for testing
+	TouchIdKeychainAccess(MacKeychain keychain) {
+		this.keychain = keychain;
+	}
+
+	@Override
+	public String displayName() {
+		return Localization.get().getString("org.cryptomator.macos.keychain.touchIdDisplayName");
+	}
+
+	@Override
+	public void storePassphrase(String key, String displayName, CharSequence passphrase) throws KeychainAccessException {
+		keychain.storePassword(SERVICE_NAME, key, passphrase, true);
+	}
+
+	@Override
+	public void storePassphrase(String key, String displayName, CharSequence passphrase, boolean requireOsAuthentication) throws KeychainAccessException {
+		keychain.storePassword(SERVICE_NAME, key, passphrase, requireOsAuthentication);
+	}
+
+	@Override
+	public char[] loadPassphrase(String key) {
+		return keychain.loadPassword(SERVICE_NAME, key);
+	}
+
+	@Override
+	public boolean isSupported() {
+		return true;
```

Comment:
This can be improved by calling native functions. See https://stackoverflow.com/questions/46887547/how-to-programmatically-check-support-of-face-id-and-touch-id and https://developer.apple.com/documentation/localauthentication/lacontext/biometrytype?language=objc

Reply:
Yes, but is this needed as we use `kSecAccessControlUserPresence` and fall back to password, when Touch ID is not available?

Restricting this to devices with Touch ID would making access to keychain items more secure for owners of non Touch ID devices impossible.

Reply:
Well, the provider is called "Touch ID".^^ Also, i'm asking myself: As a user, do i expect to always enter my login pasword on every unlock, although i clicked the "save password" checkbox?

@tobihagemann What are your thoughts?

Reply:
I agree with @infeo. Users without Touch ID should not be forced to enter the login password on every unlock.

Reply:
Done.